NDTensors/src/dense/densetensor.jl

```diff
@@ -195,7 +199,7 @@ function permutedims!(
 ) where {N,StoreT<:StridedArray}
   RA = array(R)
   TA = array(T)
-  @strided RA .= permutedims(TA, perm)
+  RA .= permutedims!!(leaf_parenttype(TA), TA, perm)
```

Comment:
I think this should be:
```julia
permutedims!!(leaf_parenttype(RA), RA, leaf_parenttype(TA), TA, perm)
```
Otherwise it won't be doing an in-place permutation, what your wrote is doing an out-of-place permutation and then a copy.